front-end/studio/package.json

```diff
@@ -12,10 +12,11 @@
     "ng": "ng",
     "start": "ng serve --port 8888 --aot",
     "debug": "ng serve --port 8888 --sourcemap -dev -v",
-    "build": "ng build -prod"
+    "build": "ng build -prod",
+    "package": "ng-packagr -p package.json"
   },
   "private": true,
-  "dependencies": {
+  "peerDependencies": {
```

Comment:
Can you explain the changes in package.json and how they will impact apicurio when built as an application (vs. as a module to be used in e.g. syndesis)?

Reply:
So this is my understanding of `dependencies` vs `peerDependencies`: when we use `dependencies` NPM (Yarn) will put the dependencies listed within, whereas with `peerDependencies` it will put them on the same level with Apicurio.

The difference can be seen in `node_modules` when you add Apicurio as a dependency:

Declared as `dependencies`:
```
node_modules
├── apicurio-design-studio
...
│   ├── node_modules
│   │   ├── bootstrap-datepicker
...
│   │   │   ├── node_modules
│   │   │   │   └── jquery
...
│   │   ├── bootstrap-switch
...
│   │   ├── brace
│   │   │   └── index.d.ts
│   │   ├── camel-case
│   │   │   └── camel-case.d.ts
...
```

Declared as `peerDependencies`:
```
├── apicurio-design-studio
...
│   ├── node_modules
│   │   ├── brace
│   │   │   └── index.d.ts
│   │   ├── camel-case
│   │   │   └── camel-case.d.ts
...
```

Notice that we then don't get `bootstrap-datepicker` or `bootstrap-switch` (and so on) dependencies in the project that uses Apicurio.

If you try to declare dependencies as `dependencies`, you get an error from ng-packagr:
```
Distributing npm packages with 'dependencies' is not recommended. Please consider adding @angular/animations to 'peerDependencies' or remove it from 'dependencies'.

BUILD ERROR
Dependency @angular/animations must be explicitly whitelisted.
Error: Dependency @angular/animations must be explicitly whitelisted.
```

This is somewhat similar to optional dependencies when using Maven.